File: Fibonacci_bignum.f90
program fibonacci_bignum
    ! Fortran implementation with string-based arbitrary precision arithmetic
    ! This implements addition of large numbers using character strings
    ! No external dependencies required
    ! Compile: gfortran Fibonacci_bignum.f90 -o fibonacci_bignum
    
    implicit none
    integer :: i, n
    character(len=500) :: t1, t2, nextTerm
    
    t1 = '0'
    t2 = '1'
    
    print *, "Enter the number of terms: "
    read *, n
    
    print *, "Fibonacci Series (using string-based arbitrary precision):"
    
    do i = 1, n
        write(*, '(A)', advance='no') trim(t1) // ", "
        call bignum_add(t1, t2, nextTerm)
        t1 = t2
        t2 = nextTerm
    end do
    
    print *
    
contains

    ! Add two positive big numbers represented as strings
    subroutine bignum_add(a, b, result)
        character(len=*), intent(in) :: a, b
        character(len=*), intent(out) :: result
        integer :: carry, sum_digit, i, j, pos
        integer :: digit_a, digit_b
        character(len=500) :: temp_result
        integer :: len_a, len_b
        
        temp_result = ''
        carry = 0
        len_a = len_trim(a)
        len_b = len_trim(b)
        i = len_a
        j = len_b
        pos = 1
        
        do while ((i > 0) .or. (j > 0) .or. (carry > 0))
            if (i > 0) then
                digit_a = ichar(a(i:i)) - ichar('0')
                i = i - 1
            else
                digit_a = 0
            end if
            
            if (j > 0) then
                digit_b = ichar(b(j:j)) - ichar('0')
                j = j - 1
            else
                digit_b = 0
            end if
            
            sum_digit = digit_a + digit_b + carry
            carry = sum_digit / 10
            temp_result(pos:pos) = char(mod(sum_digit, 10) + ichar('0'))
            pos = pos + 1
        end do
        
        ! Reverse the result
        call reverse_string(temp_result(1:pos-1), result)
    end subroutine bignum_add
    
    ! Helper subroutine to reverse a string
    subroutine reverse_string(input, output)
        character(len=*), intent(in) :: input
        character(len=*), intent(out) :: output
        integer :: i, len_input
        
        len_input = len_trim(input)
        output = ''
        
        do i = len_input, 1, -1
            output(len_input-i+1:len_input-i+1) = input(i:i)
        end do
    end subroutine reverse_string
    
end program fibonacci_bignum

File: Fibonacci_bignum.pas
program FibonacciBigNum;

{ Pascal implementation with string-based arbitrary precision arithmetic
  This implements addition of large numbers using strings
  No external dependencies required
  Compile: fpc Fibonacci_bignum.pas -ofibonacci_bignum_pas
  Run: ./fibonacci_bignum_pas }

type
    BigNumString = AnsiString;

{ Add two positive big numbers represented as strings }
function BigNumAdd(a, b: BigNumString): BigNumString;
var
    result: BigNumString;
    carry, sum, i, j, k: Integer;
    digitA, digitB: Integer;
begin
    result := '';
    carry := 0;
    i := Length(a);
    j := Length(b);
    
    while (i > 0) or (j > 0) or (carry > 0) do
    begin
        if i > 0 then
        begin
            digitA := Ord(a[i]) - Ord('0');
            Dec(i);
        end
        else
            digitA := 0;
            
        if j > 0 then
        begin
            digitB := Ord(b[j]) - Ord('0');
            Dec(j);
        end
        else
            digitB := 0;
            
        sum := digitA + digitB + carry;
        carry := sum div 10;
        result := Chr((sum mod 10) + Ord('0')) + result;
    end;
    
    BigNumAdd := result;
end;

var
    i, n: Integer;
    t1, t2, nextTerm: BigNumString;

begin
    t1 := '0';
    t2 := '1';
    
    Write('Enter the number of terms: ');
    ReadLn(n);
    
    WriteLn('Fibonacci Series (using string-based arbitrary precision):');
    
    for i := 1 to n do
    begin
        Write(t1, ', ');
        nextTerm := BigNumAdd(t1, t2);
        t1 := t2;
        t2 := nextTerm;
    end;
    
    WriteLn;
end.
